src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,286 @@
+    information about the nature of their intersection points. From that, it creates an 
+    array to represent polygon_a, and this array contains intersection points. However, after
+    calling this function, a_list is not fully formed because the neighboring indicies of the
+    intersection points in b_list still need to be updated. We will have fully formed a_list
+    after calling _build_ab_list. Also at this point we still have not update the entry and
+    exit flags for a_list. The variables poly_a and poly_b are the original polygons that the
+    user passed in. The variables edges_a and edges_b are lists of the edges of each polygon, 
+    in the form of their cartesian points. Then intr_list is a list of the intersection points
+    in their cartesion point form. Then a_idx_list is a list of the indicies of where in a_list
+    an intersection point lies. The value at index i of a_idx_list is the location in a_list where
+    the ith point of intr_list lies. Then b_idx_list indicates which EDGE of poylgon b an intersection
+    point lies. Then alpha_a_list and alpha_b_list are vectors of scalar values which help order
+    the intersection points in the case that they lie on the same edge of a polygon. Finally,
+    a_list is an empty array of PolyNode objects that gets filled out when this function is run.
+    So is b_list. 
+"""
+function _build_a_list(edges_a, edges_b,
+                      intr_list, a_idx_list, b_idx_list, alpha_a_list, alpha_b_list, a_list, b_list)
+    # Find intersection points and adds them to a_list
+    # "counter" is used to index all inter-related lists
+    counter = 1
+    # "acount" is used to index a_list
+    acount = 1
+    for ii in eachindex(edges_a)
+        # Add the first point of the edge to the list of points in a_list
+        if acount <= length(a_list)
+            a_list[acount] = PolyNode(ii, false, 0, false, 0)
+        else
+            push!(a_list, PolyNode(ii, false, 0, false, 0))
+        end
+        acount = acount + 1
+
+        # Keep track of current position in inter-related lists
+        # before finding new intersection points on our new edge
+        prev_counter = counter
+
+        for jj in eachindex(edges_b)
+
+            # Add the first point of the edge to b_list
+            if ii == 1 
+                b_list[jj] = PolyNode(jj, false, 0, false, 0)
+            end
+
+            # Check if edge jj of poly_b intersects with edge ii of poly_a
+            if _line_intersects([edges_a[ii]], [edges_b[jj]]);
+                
+                int_pt, alphas = _intersection_point(edges_a[ii], edges_b[jj])
+                # if not intersection point, skip this edge (the if statement above should 
+                # catch this but i guess not)
+                if isnothing(int_pt)
+                    continue
+                end
+                if counter <= length(intr_list)
```

Comment:
Let's just add the intersection point into `a_list` right away, rather than storing the values. Will need to check where the point fits in with respect to other intersection points on this edge (if they exist). So you probably need a counter for how many points you have added on this edge. Then, just insert/push the point in the correct position and you can delete the below section for insertion.

Reply:
You will also keep a list of the intersection indices of `a_list`. We will want to sort this by both the edge of `b` and the `beta` value. 

I think you can do this as follows:
```julia
idx = [3, 5, 7]
sort!(idx, by = x -> a_list[x].neighbor + a_list[x].fracs[2])
```
So then they should be sorted by the edge of `b` and the `beta` value since we are adding the edge (a whole number) to the edge fraction (between 0-1), which should give you the correct order to insert them I think.